Bugsnag/BugsnagSessionTracker.m

```diff
@@ -163,19 +164,30 @@ - (void)startNewSession {
         bsg_log_err(@"The session tracking endpoint has not been set. Session tracking is disabled");
         return;
     }
+    [self registerSession:[[NSUUID UUID] UUIDString] startedAt:[NSDate date] handledCount:0 unhandledCount:0];
+}
 
+- (void)registerSession:(NSString *)sessionId startedAt:(NSDate *)startedAt handledCount:(NSUInteger)handledCount unhandledCount:(NSUInteger)unhandledCount {
+    if (!sessionId || !startedAt) {
+        self.currentSession = nil;
+        BSGSessionUpdateRunContext(nil);
+        return;
```

Comment:
Why is this behaviour needed? It feels weird to clear data when a null is passed in one of two params...

Reply:
This is to continue supporting how bugsnag-unity calls the API

Reply:
Could we get a quick comment about it so that it doesn't become Chesterson's fence? :)

Reply:
After chatting to @rich-bugsnag about what bugsnag-unity uses this API for, I've tried to clean this up:

* https://github.com/bugsnag/bugsnag-cocoa/blob/6775064f61694be5badf55f6b6d25894319b337e/Bugsnag/Client/BugsnagClient.m#L449-L452
* https://github.com/bugsnag/bugsnag-unity/blob/634751779ce8ba4605c13c9796a894c504cccadc/src/BugsnagUnity.m#L365-L372

Reply:
👍